pkg/daemon/daemon.go

```diff
@@ -84,16 +79,6 @@ type TyphaDaemon struct {
 	NewClientV3           func(config apiconfig.CalicoAPIConfig) (DatastoreClient, error)
 	ConfigureEarlyLogging func()
 	ConfigureLogging      func(configParams *config.Config)
-
-	// Health monitoring.
-	healthAggregator *health.HealthAggregator
```

Comment:
Oops! You've removed too much `healthAggregator` shouldn't be removed.  

This file is run for both typha and the `typha check` command.  Only the logic that is conditional on `healthCheckOnly` should be removed.  The other logic is part of the main typha process (it's the logic for exposing typha's health on the HTTP port, which we still need).

Reply:
Ack